### ISSUE TITLE
Issues with Apple Pay?

### ISSUE THREAD

**Issue by treytt** [christianblais/Parcelify#48]
I am trying to track down a bug here and was wondering if Parcelify works with Shopify's Apple Pay implementation out of the box?

**Comment by mattlebel** [comment]
I don't believe we've seen any issues with Parcelify and Apple Pay (cia Shopify Payments) in our store, but I'd be happy to try and replicate this issue.

What's the bug behavior you've been seeing?

**Comment by mattlebel** [comment]
Just ran a few tests on my store for both domestic (US in my case) and international orders on products that have exclusively Parcelify rates. I didn't run into any "We don't currently ship to your country" errors when trying with Apple Pay.

Just talking about loud about how I would debug this:

0. Clone your setup onto a staging/test store. You can create a free [development store](https://help.shopify.com/partners/dashboard/development-stores) or, if you are on Shopify Plus, have your MSM clone your existing store for you.

1. Temporarily disable all Parcelify shipping rules for the country you are trying to ship to. (The toggle to disable individual shipping rules is under the "Advanced Settings" section of each respective zone.)

Additionally, delete/disable any other 3rd party calculated shipping apps you may have installed.

2. Go into the main Settings -> Shipping area of your Shopify dashboard and look at your shipping zones. 

Do you have a shipping zone for the country you are trying to ship to? If so, do you have a shipping region (e.g. State or Province) setup for the country you are trying to ship to?

_For Shopify Plus merchants_, make sure you don't have any Line item scripts or checkout.liquid code that could be interfering with desired shipping rates.

3. Once you've verified that your tests orders with Apple Pay work with the standard shipping settings in Shopify (ruling out any kind of platform-wide errors or forgetting to include country/region in available shipping regions), I would love into Parcelify debugging:

- Check the _weights and prices_ on your products and make sure you're not disqualifying orders based on "Limits"

<img width="1003" alt="screen shot 2018-04-23 at 10 37 10 pm" src="https://user-images.githubusercontent.com/1132438/39168001-f1f740d6-4746-11e8-8ee6-b628a7e00395.png">

- Check your _Filters_ -- are you disqualifying any regions based on exceptions? For example, have you excluded Alaska and Hawaii from your main shipping zone and forgot to make a separate _inclusive_ zone for them?

- Check your _ISO country code_ abbreviations. Have you done something like: used "UK" instead of "GB" for the United Kingdom, used "IR" instead of "IE" for Ireland, etc.

<img width="977" alt="screen shot 2018-04-23 at 10 38 06 pm" src="https://user-images.githubusercontent.com/1132438/39168074-4cff4b7c-4747-11e8-93b4-00872e70fc81.png">

- If you're using _RegEx in your Filters_, test out the expression in isolation using a tool like https://regexr.com. Do your test cases pass?

4. If you are still stuck, take a higher level look at all your Parcelify shipping zones -- is anything overlapping? Are any rules contradicting each other?

Hope this helps!
